src/compiler/checker.ts

```diff
@@ -2533,10 +2533,24 @@ namespace ts {
 
         /* Starting from 'initial' node walk up the parent chain until 'stopAt' node is reached.
          * If at any point current node is equal to 'parent' node - return true.
+         * If current node is an IIFE, continue walking up.
          * Return false if 'stopAt' node is reached or isFunctionLike(current) === true.
          */
         function isSameScopeDescendentOf(initial: Node, parent: Node | undefined, stopAt: Node): boolean {
-            return !!parent && !!findAncestor(initial, n => n === stopAt || isFunctionLike(n) ? "quit" : n === parent);
+            return !!parent && !!findAncestor(initial, n => {

```

Comment:
Isn't this equivalent? I *think* I like it better simply because it's so much shorter, although I'm not 100% sure.
```suggestion
            return !!parent && !!findAncestor(initial, n => n === stopAt || isFunctionLike(n) && !getImmediatelyInvokedFunctionExpression() : n === parent);
```

Reply:
Yeah, it works. But I wonder if we should have `n === parent` before `isFunctionLike(n) && !getImmediatelyInvokedFunctionExpression()` because `n` might be a function.